staging/src/k8s.io/client-go/tools/remotecommand/websocket.go

```diff
@@ -65,8 +61,8 @@ const (
 	// "pong" message before a timeout error occurs for websocket reading.
 	// This duration must always be greater than the "pingPeriod". By defining
 	// this deadline in terms of the ping period, we are essentially saying
-	// we can drop "X-1" (e.g. 3-1=2) pings before firing the timeout.
-	pingReadDeadline = (pingPeriod * 3) + (1 * time.Second)
+	// we can drop "X-1" (e.g. 6-1=5) pings before firing the timeout.
+	pingReadDeadline = (pingPeriod * 6) + (1 * time.Second)
```

Comment:
This means we'll push the read deadline out ~31 seconds into the future every time we read a pong response (which ~should come every ~5 seconds). As soon as that deadline gets hit, reads to the connection will always return error. I think this is our "how long do we wait on a dead TCP connection, to send a ping or receive a ping", and use it as the read deadline and write deadline in the ping handler.